tweet_recommendations/recommendation.py

```diff
@@ -11,32 +11,46 @@
 
 
 def embedding_similarity(x: np.ndarray, y: np.ndarray):
-    similarity = cosine_similarity(x, y)
+    similarity = np.clip(cosine_similarity(x, y), -1, 1)
     ang_dist = np.arccos(similarity) / np.pi
     ang_sim = 1 - ang_dist
     return ang_sim
 
 
 def normalise(array: np.ndarray):
-    return (array - array.min()) / (array.max() - array.min())
+    if array.max() != array.min():
+        return (array - array.min()) / (array.max() - array.min())
+    else:
```

Comment:
Ten else chyba nie jest potrzebny?

Btw ta normalizacja powinna być w przedziale [0, 1]? Jeśli jednak [-1, 1] to niżej mamy sampla: 

> array = array - array.mean()
>array = array / array.max()

Reply:
Nie wiem chyba dlaczego normalizacja miałaby być do [-1, 1]...
Else wyrzucony